### PR TITLE
Netsuite export settings

### PR DESCRIPTION
**Pull request by NileshPant1999**
## Summary by CodeRabbit

- **New Features**
	- Introduced NetSuite export settings management for improved integration.
	- Added help article links for NetSuite settings in the app branding configuration.
	- Enhanced NetSuite onboarding with structured layout and dynamic settings components.
- **Enhancements**
	- Updated workspace settings fetch method for better performance.
	- Reordered imports in tracking service for consistency.
	- Revamped NetSuite export settings UI with conditional rendering and added styling.
- **Refactor**
	- Streamlined NetSuite integration components and shared module organization.
	- Removed unused imports and components for cleaner codebase.
- **Bug Fixes**
	- Corrected the options property in QBO clone settings for accurate account mapping.

### CODE REVIEW

**Comment by NileshPant1999** [comment]
@coderabbitai review

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.html [hunk 1]
```diff
@@ -1 +1,235 @@
...
+                                [iconPath]="'list'"
+                                [placeholder]="'Choose the type of transaction in NetSuite to export your ' + brandingConfig.brandName +' expenses'"
+                                [formControllerName]="'reimbursableExportType'"
+                                [appName]="appName"
+                                [showExportPreview]="true"
+                                [exportTypeIconPathArray]="previewImagePaths">
+                            </app-configuration-select-field>
+                        </div>
+
+                        <div *ngIf="exportSettingForm.value.reimbursableExportType" class="tw-mt-16-px tw-bg-white tw-border tw-border-solid tw-border-separator tw-rounded-12-px">
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.html:46]: we should get employee field mapping only for JE. If ER is selected, patch value to EMPLOYEE. If Bill is selected patch value to VENDOR.

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.html [hunk 2]
```diff
@@ -1 +1,235 @@
...
+                            <ng-container *ngTemplateOutlet="employeeFieldMapping"></ng-container>
+                            <div *ngIf="exportSettingForm.value.reimbursableExportType">
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.html:48]: condition needs to change

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    @apply tw-border-alert-toast #{!important};
+}
+
+:host ::ng-deep .normal-box  {
+    @apply tw-border-box-color #{!important};
+}
+
+:host ::ng-deep .normal-box :enabled:focus-visible {
+    @apply tw-border-slightly-normal-text-color #{!important};
+}
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.scss:11]: all these are not used, remove

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+.sub-text {
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.scss:13]: remove this. Use tw-text-text-muted wherever this is used

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts [hunk 5]
```diff
@@ -7,9 +25,234 @@ import { Component, OnInit } from '@angular/core';
...
+
+  accountsPayables: DefaultDestinationAttribute[];
+
+  vendors: DefaultDestinationAttribute[];
+
+  isImportItemsEnabled: boolean;
+
+  reimbursableExportTypes: SelectFormOption[];
+
+  autoMapEmployeeOptions: ExportSettingFormOption[] = [
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts:54]: get from model

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts [hunk 6]
```diff
@@ -7,9 +25,234 @@ import { Component, OnInit } from '@angular/core';
...
+    { label: 'Based on Employee E-mail ID', value: 'EMAIL' },
+    { label: 'Based on Employee Name', value: 'NAME' },
+    { label: 'Based on Employee Code', value: 'EMPLOYEE_CODE' }
+  ];
+
+  employeeFieldOptions: ExportSettingFormOption[] = [
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts:60]: get from model

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts [hunk 7]
```diff
@@ -7,9 +25,234 @@ import { Component, OnInit } from '@angular/core';
...
+
+      if (isreimbursableExportTypeSelected === NetsuiteReimbursableExpensesObject.BILL) {
+        this.exportSettingForm.controls.employeeFieldMapping.patchValue(FyleField.VENDOR);
+        this.exportSettingForm.controls.employeeFieldMapping.disable();
+      }
+    });
+  }
+
+  private createReimbursableExpenseWatcher(): void {
+    this.exportSettingForm.controls.reimbursableExpense.valueChanges.subscribe((isReimbursableExpenseSelected) => {
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts:159]: do validations in sage 300 / dynamics style, we've generalised a bunch of things.

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts [hunk 8]
```diff
@@ -7,9 +25,234 @@ import { Component, OnInit } from '@angular/core';
...
+      const exportSettingPayload = NetSuiteExportSettingModel.constructPayload(this.exportSettingForm);
+      this.exportSettingService.postExportSettings(exportSettingPayload).subscribe((response: NetSuiteExportSettingGet) => {
+        this.isSaveInProgress = false;
+        this.toastService.displayToastMessage(ToastSeverity.SUCCESS, 'Export settings saved successfully');
+
+        if (this.isOnboarding) {
+          this.workspaceService.setOnboardingState(NetsuiteOnboardingState.IMPORT_SETTINGS);
+          this.router.navigate([`/integrations/netsuite/onboarding/import_settings`]);
+        } else {
+          this.router.navigate(['/integrations/netsuite/main/dashboard']);
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts:215]: remove else case

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts [hunk 9]
```diff
@@ -7,9 +25,234 @@ import { Component, OnInit } from '@angular/core';
...
+  }
+
+  private getSettingsAndSetupForm(): void {
+    this.isOnboarding = this.windowReference.location.pathname.includes('onboarding');
+    forkJoin({
+      exportSetting: this.exportSettingService.getExportSettings()
+    }).subscribe(({exportSetting}) => {
+      this.exportSettings = exportSetting;
+
+      this.reimbursableExportTypes = NetSuiteExportSettingModel.getReimbursableExportTypeOptions();
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts:239]: do it at top where you declare variable

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts [hunk 10]
```diff
@@ -7,9 +25,234 @@ import { Component, OnInit } from '@angular/core';
...
+      this.showNameInJournalOption = this.exportSettings.configuration?.corporate_credit_card_expenses_object === NetSuiteCorporateCreditCardExpensesObject.JOURNAL_ENTRY ? true : false;
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts:240]: on form changes this value should change

File: src/app/core/models/netsuite/netsuite-configuration/netsuite-export-setting.model.ts [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+import { ExportSettingFormOption } from "../../intacct/intacct-configuration/export-settings.model";
+import { disable } from "@rxweb/reactive-form-validators";
+
+
+export type NetsuiteExportSettingWorkspaceGeneralSettingPost = {
+    employee_field_mapping: EmployeeFieldMapping,
+    reimbursable_expenses_object: NetsuiteReimbursableExpensesObject | null,
+    auto_map_employees: ExportSettingFormOption | null,
+    corporate_credit_card_expenses_object: NetSuiteCorporateCreditCardExpensesObject | null
+    name_in_journal_entry: NameInJournalEntry;
```
[the remark this answers is not in the document]
Comment (NileshPant1999) [reply src/app/core/models/netsuite/netsuite-configuration/netsuite-export-setting.model.ts:17]: sure will add

File: src/app/core/models/netsuite/netsuite-configuration/netsuite-export-setting.model.ts [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+          cccExpenseState: new FormControl(exportSettings?.expense_group_settings?.ccc_expense_state),
+          creditCardExpense: new FormControl(exportSettings?.configuration?.corporate_credit_card_expenses_object ? true : false),
+          creditCardExportType: new FormControl(exportSettings?.configuration?.corporate_credit_card_expenses_object),
+          creditCardExportGroup: new FormControl(this.getExportGroup(exportSettings?.expense_group_settings?.corporate_credit_card_expense_group_fields)),
+          creditCardExportDate: new FormControl(exportSettings?.expense_group_settings?.ccc_export_date_type),
+          defaultCCCAccount: new FormControl(exportSettings?.general_mappings?.default_ccc_account?.id ? exportSettings.general_mappings.default_ccc_account : null),
+          accountsPayable: new FormControl(exportSettings?.general_mappings?.accounts_payable?.id ? exportSettings.general_mappings.accounts_payable : null),
+          defaultCreditCardVendor: new FormControl(exportSettings?.general_mappings?.default_ccc_vendor?.id ? exportSettings.general_mappings.default_ccc_vendor : null)
```
[the remark this answers is not in the document]
Comment (NileshPant1999) [reply src/app/core/models/netsuite/netsuite-configuration/netsuite-export-setting.model.ts:244]: this is fixed in second pr, will get resolved automatically when that is merged

File: src/app/core/models/netsuite/netsuite-configuration/netsuite-export-setting.model.ts [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+          },
+          configuration: {
+            reimbursable_expenses_object: exportSettingsForm.get('reimbursableExportType')?.value,
+            corporate_credit_card_expenses_object: exportSettingsForm.get('creditCardExportType')?.value,
+            employee_field_mapping: exportSettingsForm.get('employeeFieldMapping')?.value,
+            auto_map_employees: exportSettingsForm.get('autoMapEmployees')?.value,
+            name_in_journal_entry: nameInJournalEntry
+          },
+          general_mappings: {
+            reimbursable_account: exportSettingsForm.get('defaultCCCAccount')?.value ? exportSettingsForm.get('defaultCCCAccount')?.value : emptyDestinationAttribute,
```
[the remark this answers is not in the document]
Comment (NileshPant1999) [reply src/app/core/models/netsuite/netsuite-configuration/netsuite-export-setting.model.ts:275]: fixed in next pr

File: src/app/core/models/netsuite/netsuite-configuration/netsuite-export-setting.model.ts [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+            default_ccc_account: exportSettingsForm.get('accountsPayable')?.value ? exportSettingsForm.get('accountsPayable')?.value : emptyDestinationAttribute,
+            accounts_payable: exportSettingsForm.get('defaultCreditCardVendor')?.value ? exportSettingsForm.get('defaultCreditCardVendor')?.value : emptyDestinationAttribute,
```
[the remark this answers is not in the document]
Comment (NileshPant1999) [reply src/app/core/models/netsuite/netsuite-configuration/netsuite-export-setting.model.ts:277]: same

File: src/app/core/models/netsuite/netsuite-configuration/netsuite-export-setting.model.ts [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+            default_ccc_vendor: exportSettingsForm.get('defaultDebitCardAccount')?.value ? exportSettingsForm.get('defaultDebitCardAccount')?.value : emptyDestinationAttribute
```
[the remark this answers is not in the document]
Comment (NileshPant1999) [reply src/app/core/models/netsuite/netsuite-configuration/netsuite-export-setting.model.ts:278]: same

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.html (hunk 1 above)
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.html:46]: this is not fixed yet

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.html (hunk 2 above)
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.html:48]: not fixed yet

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.scss [hunk 16]
```diff
@@ -1,20 +0,0 @@
...
-}
-
-.sub-text {
-    color: #5A5D72;
-    font-size: 12px;
-}
-
-.radius {
-    border-radius: 4px;
-}
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.scss]: these 2 are used in html

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts [hunk 17]
```diff
@@ -7,9 +25,205 @@ import { Component, OnInit } from '@angular/core';
...
+  ) {
+    this.windowReference = this.windowService.nativeWindow;
+  }
+
+
+  refreshDimensions() {
+    this.netsuiteHelperServie.refreshNetsuiteDimensions().subscribe();
+  }
+
+  private reimbursableExportTypeWatcher(): void {
```
Comment (NileshPant1999) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts:125]: here @ashwin1111

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.scss (hunk 4 above)
Comment (ashwin1111) [reply src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.scss:1]: needs fixing

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**